File: BaekJoon Online Judge/C++/Step09-Math_2 Prime and Geometric/2581.cpp
//
//  2581.cpp
//  Step09-Math_2 Prime and Geometric
//
//  Created by Yebin Kim on 2019/10/24.
//  Copyright © 2019 김예빈. All rights reserved.
//

#include <iostream>

using namespace std;

bool searchPrime(int);

// Time Complexity O((n-m)*log(n))
int main() {
    cin.tie(NULL);
    ios::sync_with_stdio(false);
    
    int m, n, sum = 0, min = 10000;
    
    cin >> m >> n;
    
    for(int i = m; i <= n; i++) {
        if(searchPrime(i)) {
            sum += i;
            
            if(i < min)
                min = i;
        }
    }
    
    if(sum != 0)
        cout << sum << "\n" << min;
    else
        cout << "-1";
    
    return 0;
}

bool searchPrime(int input) {
    if(input < 2) return false;
    
    for(int i = 2; i * i <= input; i++)
        if(input % i == 0)
            return false;
    
    return true;
}

File: BaekJoon Online Judge/Swift/Step09-Math_2 Prime and Geometric/2581.swift
//
//  2581.swift
//  Step09-Math_2 Prime and Geometric
//
//  Created by Yebin Kim on 2019/11/17.
//  Copyright © 2019 김예빈. All rights reserved.
//

import Foundation

func findPrime(_ num: Int) -> Bool {
    if num < 2 {
        return false
    }
    
    var i = 2
    
    while i * i <= num {
        if num % i == 0 {
            return false
        }
        
        i += 1
    }
    
    return true
}

if let M = Int(readLine()!), let N = Int(readLine()!) {
    let input = stride(from: M, to: N + 1, by: 1).filter{ findPrime($0) == true }
    let sum = input.reduce(0, { return $0 + $1 })
    
    if input.count == 0 {
        print(-1)
    } else {
        print(sum)
        print(input.min()!)
    }
}
